crates/weaver_resolver/README.md

```diff
@@ -33,14 +35,80 @@ involves the following steps:
+    found.
 - Apply constraints `any_of` and `include`.
 - Validate the resolved semantic conventions
   - No more unresolved `ref` or `extends` clauses. The unresolved list should
     be empty.
   - All constraints satisfied.
 
+## Lineage
+
+The resolution process can optionally compute the lineage for each attribute of
+a semantic convention registry. The lineage as such is not part of the syntax
+and structure of a semantic convention; rather, it's an extension produced by
+the `weaver` tool, intended for use in scenarios such as:
+- A semconv author wishes to verify the exact path followed by the resolution
+process in the case of a complex cascade of inheritance across multiple levels
+between groups.
+- A documentation process aims to add lineage information to the documentation
+of each attribute to allow readers to understand where the definitions of each
+field were defined.
+- The lineage information of a semantic convention could eventually feed into
+an enterprise data catalog to improve the data governance process.
+
+The general structure of the lineage generated by the resolution process is as
+follows:
+
+```json
+{
+  "registry_url": "<registry url>",
+  "groups": [
+    {
+      // Standard semconv fields 
+      "id": "<group id>",
+      "type": "<group type>",
+      "brief": "<group brief>",
+      "attributes": [ /* ... */ ],
+      "lineage": {
+        // `source_file` is either:
+        // - a relative path in the case of a registry that has been explicitly
+        // resolved from a directory in the file system. This type of registry
+        // is mainly used in unit tests, where using a relative path makes
+        // these tests independent of where they are executed.
+        // - an absolute URL when the resolution applies to a registry
+        // accessible from a GitHub repository, for example.
+        "source_file": "<path or url>",
+        "attributes": {
+          "<attribute id>": {
+            // The group ID where the attribute fields were defined before a
+            // possible local redefinition. In the case of an inherited
+            // attribute, this field contains the ID of the group defined in
+            // the `extends` clause of the initial definition. In the case of
+            // a reference to a root attribute, this field contains the ID of
+            // the group where this attribute was defined globally.
+            "source_group": "<group id>",
+            // The field names inherited from the source group. This field is
+            // present only if the attribute is inherited.
+            "inherited_fields": [ "<field name>", /* ... */ ],
+            // The field names overridden in the local group.
+            "locally_overridden_fields": [ "<field name>", /* ... */ ],
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+Important note: Fields that appear in the group but are not mentioned in the
```

Comment:
This means the group is the *first* time they appear?  Is this why you split inherited from overridden?

If we have a simple "local always overrides" I think perhaps maybe just "locally_defined" is enough, as anything locally defined would override inherited and you can (easily?) recompute inherited fields by following the lineage train.

Reply:
I replaced "Fields that appear ..." with "Attributes that appear ..." because it was ambiguous. I was talking about attributes that are totally defined locally in a group (i.e. attribute with an ID).